aiida/orm/nodes/data/array/xy.py

```diff
@@ -43,19 +43,19 @@ class XyData(ArrayData):
     def _arrayandname_validator(array, name, units):
         """
         Validates that the array is an numpy.ndarray and that the name is
-        of type str. Raises InputValidationError if this not the case.
+        of type str. Raises TypeError or ValueError if this not the case.
         """
         if not isinstance(name, str):
-            raise InputValidationError('The name must always be a str.')
+            raise TypeError('The name must always be a str.')
 
         if not isinstance(array, np.ndarray):
-            raise InputValidationError('The input array must always be a numpy array')
+            raise TypeError('The input array must always be a numpy array')
         try:
             array.astype(float)
         except ValueError:
-            raise InputValidationError('The input array must only contain floats')
+            raise ValueError('The input array must only contain floats')
```

Comment:
Nitpick, but wouldn't this be a `TypeError`?

Reply:
Yeah, I also was wondering about this, but since the actual top level type (list) was correct, I turned it into a `ValueError`. But thinking more about it I think you are right. They type should be `List[float]` and so the type of the list elements is part of the entire type, which is not respected. Will change it.

Reply:
Yes, I also hesitated a bit on this one, t'was a tricky one.